server/datastore/mysql/hosts.go

```diff
@@ -984,6 +990,17 @@ func (d *Datastore) DeleteHosts(ctx context.Context, ids []uint) error {
 	if err != nil {
 		return ctxerr.Wrap(ctx, err, "deleting host emails")
 	}
+
+	query, args, err = sqlx.In(`DELETE FROM pack_targets WHERE type=? AND target_id in (?)`, fleet.TargetHost, ids)
+	if err != nil {
+		return ctxerr.Wrapf(ctx, err, "building delete pack_targets query")
+	}
+
+	_, err = d.writer.ExecContext(ctx, query, args...)
+	if err != nil {
+		return ctxerr.Wrapf(ctx, err, "deleting pack_targets for hosts")
+	}
+
 	return nil
```

Comment:
Just like in `DeleteHost`, maybe we should wrap `DeleteHosts` on a transaction? (though it could be a big transaction...)

Reply:
Hm, I feel like `delete ... where id in ...` will be faster than thosands of `delete ... where id = ...`.